cmd/snap-confine/snap-confine.apparmor.in

```diff
@@ -492,6 +492,9 @@
         mount options=(rw bind) /var/snap/*/** -> /var/snap/*/**,
         mount options=(ro bind) /var/snap/*/** -> /var/snap/*/**,
 
+        # Allow creating missing mount directories under $SNAP_DATA
+        /var/snap/*/** rw,
```

Comment:
To limit to directories, use `/var/snap/*/**/ w,` (do you actually need 'r'?).

Reply:
We use stat so I assumed we do.

Reply:
I'll come up with a few spread tests and if we don't need this I will remove this.

Reply:
We need the `r` for the SecureMkdirAll implementation now.